### PR TITLE
Add support for Caddy (#1768)

### LINKED ISSUE #1768: [Icon Request] Support for Caddy
- [x] I'm sure this issue is not a _duplicate_.
- [x] I'm submitting an icon request:
  - Type: `languageId`
  - Icon Name: `caddyserver`
  - Sample original Icon: logo
  - LAnguageId: `caddyfile`

- Language Extensions: https://github.com/Zamerick/vscode-caddyfile-syntax
- More info: https://caddyserver.com/

### PR DESCRIPTION
**Pull request by nitroin**
_**Fixes #1768 **_

**Changes proposed:**

- [x] Add
- [ ] Delete
- [ ] Fix
- [ ] Prepare

**Notes:** given that I didn't found an official `.svg` resource I used the `favicon.ico` from the official site: resized to 28x28 (2 px margin for total 32x32).

### CODE REVIEW

File: src/icon-manifest/supportedExtensions.ts [hunk 1]
```diff
@@ -579,6 +579,13 @@ export const extensions: IFileCollection = {
       languages: [languages.cabal],
       format: FileFormat.svg,
     },
+    {
+      icon: 'caddy',
+      extensions: ['caddyfile'],
```
Comment (JimiC) [inline src/icon-manifest/supportedExtensions.ts]: Since you added `languages`, `extensions` assignment is redundant.

Comment (nitroin) [reply src/icon-manifest/supportedExtensions.ts]: That wouldn't drop support for those without the 3rd party extension?

Comment (JimiC) [reply src/icon-manifest/supportedExtensions.ts]: Yes and that's our policy.

**Comment by nitroin** [comment]
Yup, before doing it myself I've requested for an official asset: 

https://github.com/mholt/caddy/issues/2335

I'll wait 1 day or 2 before DIY.

**Comment by nitroin** [comment]
Hi, there! Just an update... items worked since the last review:

- [x] drop `extensions`
- [x] use `languages ` only
- [x] use `svg` instead of `png`

**Pull request merged**